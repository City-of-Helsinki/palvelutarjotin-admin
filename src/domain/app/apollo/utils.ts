import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
