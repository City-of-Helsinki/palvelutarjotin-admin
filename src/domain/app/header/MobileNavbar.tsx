import { IconClose, IconMenu } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

import Container from '../layout/Container';
import MobileMenu, { useMobileMenuContext } from '../mobileMenu/MobileMenu';
import styles from './mobileNavbar.module.scss';

const MobileNavbar: React.FC = () => {
  const { t } = useTranslation();
  const {
    closeMobileMenu,
    isMobileMenuOpen,
    openMobileMenu,
  } = useMobileMenuContext();

  const handleCloseMenu = () => {
    closeMobileMenu();
  };

  const handleOpenMenu = () => {
    openMobileMenu();
  };

  return (
    <>
      <Container>
        <div className={styles.mobileNavbar}>
          <Link
            aria-label={t('header.ariaLabelLogo')}
            onClick={handleCloseMenu}
            to={'/'}
            className={styles.logoWrapper}
          >
            <div className={styles.logo} />
            <div className={styles.appName}>{t('appName')}</div>
          </Link>
          <div className={styles.buttonWrapper}>
            {isMobileMenuOpen ? (
              <button
                className={styles.closeButton}
                onClick={handleCloseMenu}
                aria-label={t('header.ariaButtonCloseMenu')}
              >
                <IconClose />
              </button>
            ) : (
              <button
                className={styles.menuButton}
                onClick={handleOpenMenu}
                aria-label={t('header.ariaButtonOpenMenu')}
              >
                <IconMenu />
              </button>
            )}
          </div>
        </div>
      </Container>
      <MobileMenu isMenuOpen={isMobileMenuOpen} onClose={handleCloseMenu} />
    </>
  );
};

export default MobileNavbar;
