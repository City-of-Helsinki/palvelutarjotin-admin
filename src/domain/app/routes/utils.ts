import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
