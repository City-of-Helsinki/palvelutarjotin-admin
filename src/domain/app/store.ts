import { configureStore, getDefaultMiddleware, Store } from '@reduxjs/toolkit';
import { loadUser } from 'redux-oidc';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import userManager from '../auth/userManager';
import reducer from './reducers';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['organisation'],
};

const persistedReducer = persistReducer(persistConfig, reducer);

const store: Store = configureStore({
  devTools: true,
  middleware: getDefaultMiddleware({
    serializableCheck: false,
  }),
  reducer: persistedReducer,
});

loadUser(store, userManager);
const persistor = persistStore(store);

export { persistor, store };

// Define a pretyped dispatcher and root state: redux.js.org/usage/usage-with-typescript#define-typed-hooks
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
