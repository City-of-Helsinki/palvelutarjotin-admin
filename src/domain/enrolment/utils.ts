import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
