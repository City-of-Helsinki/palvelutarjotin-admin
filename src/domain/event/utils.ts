import { SUPPORT_LANGUAGES } from '../../constants';
import { EventQuery } from '../../generated/graphql';
import { Language } from '../../types';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';
/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

/**
 * Get first availanle language of an event
 * @param {object} eventData
 * @return {string}
 */
export const getFirstAvailableLanguage = (eventData: EventQuery): Language => {
  if (eventData.event?.name.en) return SUPPORT_LANGUAGES.EN;
  if (eventData.event?.name.fi) return SUPPORT_LANGUAGES.FI;
  if (eventData.event?.name.sv) return SUPPORT_LANGUAGES.SV;

  return SUPPORT_LANGUAGES.FI;
};
