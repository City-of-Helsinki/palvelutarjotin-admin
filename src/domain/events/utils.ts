import { isFuture } from 'date-fns';
import forEach from 'lodash/forEach';

import { EventFieldsFragment } from '../../generated/graphql';

export const hasOccurrences = (event: EventFieldsFragment) => {
  return Boolean(event.pEvent?.occurrences.edges.length);
};

export const hasComingOccurrences = (event: EventFieldsFragment) => {
  let hasComingItems = false;

  forEach(event.pEvent?.occurrences.edges, (edge) => {
    if (edge?.node?.startTime && isFuture(new Date(edge?.node?.startTime))) {
      hasComingItems = true;
      return false;
    }
  });

  return Boolean(hasComingItems);
};
