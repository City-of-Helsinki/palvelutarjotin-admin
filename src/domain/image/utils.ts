import { ImageDocument, ImageQuery } from '../../generated/graphql';
import { initializeApolloClient } from '../app/apollo/apolloClient';

export const getImageName = (id: string) => {
  const apolloClient = initializeApolloClient();
  const data = apolloClient.readQuery<ImageQuery>({
    query: ImageDocument,
    variables: { id },
  });
  return data?.image?.name;
};
