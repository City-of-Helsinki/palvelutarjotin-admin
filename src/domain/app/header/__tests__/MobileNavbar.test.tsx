import { MockedProvider } from '@apollo/react-testing';
import { render } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';

import { MyProfileDocument } from '../../../../generated/graphql';
import { fakePerson } from '../../../../utils/mockDataUtils';
import { store } from '../../store';
import MobileNavbar from '../MobileNavbar';

const profileResponse = {
  data: {
    myProfile: fakePerson(),
  },
};

const mocks = [
  {
    request: {
      query: MyProfileDocument,
    },
    result: profileResponse,
  },
];

it('MobileNavbar matches snapshot', () => {
  const { container } = render(
    <MockedProvider mocks={mocks}>
      <Provider store={store}>
        <MemoryRouter>
          <MobileNavbar />
        </MemoryRouter>
      </Provider>
    </MockedProvider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
