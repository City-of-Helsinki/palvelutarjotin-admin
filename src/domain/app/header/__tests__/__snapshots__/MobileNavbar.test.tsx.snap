// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavbar matches snapshot 1`] = `"<div class=\\"container\\"><div class=\\"mobileNavbar\\"><a aria-label=\\"Etusivu\\" class=\\"logoWrapper\\" href=\\"/\\"><div class=\\"logo\\"></div><div class=\\"appName\\">Palvelutarjotin Admin</div></a><div class=\\"buttonWrapper\\"><button class=\\"menuButton\\" aria-label=\\"Avaa valikko\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><path d=\\"M1.01301075,4.01419355 L22.9875806,4.01419355 L22.9875806,6.61634409 L1.01301075,6.61634409 L1.01301075,4.01419355 Z M1.01301075,17.7482258 L22.9875806,17.7482258 L22.9875806,20.3503763 L1.01301075,20.3503763 L1.01301075,17.7482258 Z M1.01301075,10.8815054 L22.9875806,10.8815054 L22.9875806,13.4836559 L1.01301075,13.4836559 L1.01301075,10.8815054 Z\\"></path></svg></button></div></div></div><div aria-hidden=\\"true\\" class=\\"mobileMenu\\" style=\\"visibility:hidden\\"><div class=\\"linkWrapper\\"><ul></ul></div><div class=\\"languageSelectWrapper\\"><ul><li class=\\"languageLink isSelected\\"><a lang=\\"fi\\" href=\\"/\\">Suomeksi</a></li><li class=\\"languageLink\\"><a lang=\\"sv\\" href=\\"/\\">PÃ¥ svenska</a></li><li class=\\"languageLink\\"><a lang=\\"en\\" href=\\"/\\">In English</a></li></ul></div></div>"`;
